### PR TITLE
Decision making and branching in java

### LINKED ISSUE #2348: Decision making and branching in java
### Proposed title of article
Decision making and branching in java

### Introduction paragraph (2-3 paragraphs):
A java program is a set of statements, which are normally executed sequentially in the order in which they appear. This happens when options or repetitions of certain calculations are not necessary. However, in practice, we have a number of situations, where we may have to change the order of the execution of statements based on certain conditions, or repeat a group of statements until certain specified conditions are met. This involves a kind of decision making to see whether a particular condition has occurred or not and then direct the computer to execute certain statements accordingly.
When a program breaks the sequential flow and jumps to another part of the code, it is called branching. When the branching is based on a particular condition, it is known as conditional branching. If branching takes place without without any   decision, it is known as unconditional branching.
Java language possesses such decision making capabilities and supports control or decision making statements.

### Key takeaways
1.	If statement
2.	If…else statement
3.	Nested if…else statement
4.	Switch statement

### References: 
N/A

### PR DESCRIPTION
**Pull request by Georgewamama**
### Article checklist

**ATTENTION: In your PR - add the ISSUE (topic form) # that this PR corresponds to. You can do this by also adding a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.**
      
      
**NOTE: (Please ensure that you have only one open issue + linked pull request at a time. This will ensure that we complete the article in a timely manner from inception to publishing.)**

**If you have not already please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve our overall technical writing.**

#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [ ] Please place your article through a 3rd party plagiarism checker? We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [ ] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**

This fixes #2348

### CODE REVIEW

**Comment by hectorkambow** [comment]
@Georgewamama 
Good afternoon and welcome to the EngEd program - thank you for submitting your article for publishing. It has been entered into our queue and should be reviewed (and published pending any changes) as soon as a peer reviewer is finished with the ones in the queue before it.  🚀

File: content/articles/decision-making-and branching-in-java/index.md [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+  System.out.println("First upper.");
+} else if (marks > 59) {
+  System.out.println("Second upper.");
+} else if(marks > 49) {
+  System.out.println("Second lower.");
+} else if(marks < 49){
+System.out.println("fail");
+}
+  }
+}
```
Comment (WanjaMIKE) [inline content/articles/decision-making-and branching-in-java/index.md:208]: The above program will work well if structured as shown below:

```
public class Main {
  public static void main(String[] args) {
   int marks = 22;

    if (marks >= 79) { //if a students marks are greater than 79
      System.out.println("First upper."); 
    } else if (marks >= 59 && marks < 79) { //if marks are greater than 59 but less than 79
      System.out.println("Second upper.");
    } else if(marks >= 49 && marks < 59) { //if marks are greater than 49 but less than 59
      System.out.println("Second lower.");
    } else if(marks < 49){
    System.out.println("fail");
    }
  }
}
```
I have made the adjustments

**Comment by hectorkambow** [comment]
@Georgewamama 

👋 Good afternoon - and welcome to the EngEd program.
I'll be running a final review on your PR when ready.
Upon running your article through our 3rd party plagiarism checker it seemed to raise a few flags and the % was higher than we typically accept.
Please see attached PDF - and revisit the article to ensure we are contributing wholly unique and original content.
Be sure to see our resources page to see more info on plagiarism and what is considered as such.


[Decision making and branching in java #2739.pdf](https://github.com/section-engineering-education/engineering-education/files/6802615/Decision.making.and.branching.in.java.2739.pdf)

**Comment by Georgewamama** [comment]
Hello
 @hectorkambow and @WanjaMIKE updated the changes towards the article and consider checking on the plagiarism % again

**Comment by ahmadmardeni1** [comment]
@Georgewamama 
This seems to be an interesting article. However, there are a lot of articles covering "if-else" and "switch" and we can't publish this article as it is because we are looking for unique/original content. Therefore, you should improve the article by talking about the use cases of "if-else" and "case", highlight real-life use cases, where they should use them, nested if-else vs switch, etc...
You can add any value you want (The above examples are just for your reference) to improve it and make it different from what can be found easily on the internet.

I will close this PR - when you modify it we can re-open it at any time.

Please reach your peer reviewer or me at any time if you have any questions.
Cc @WanjaMIKE

**Comment by Georgewamama** [comment]
Hello @ahmadmardeni1 and @WanjaMIKE 
I have worked on the PR and made advanced changes as requested
consider looking at it

**Comment by ahmadmardeni1** [comment]
Hello @Georgewamama, you didn't commit the changes.

**Comment by Georgewamama** [comment]
@ahmadmardeni1 committed the changes

**Comment by ahmadmardeni1** [comment]
File renamed without changes. @Georgewamama

**Comment by Georgewamama** [comment]
@ahmadmardeni1 
have a look at the new update

**Comment by ahmadmardeni1** [comment]
Great article, and thank you for the enhancement 🚀 @Georgewamama

**Pull request merged**